docs/source/notes/extending.rst

```diff
@@ -1,10 +1,238 @@
 Extending PyTorch
 =================
 
-In this note we'll cover ways of extending :mod:`torch.nn`,
+In this note we'll cover ways of extending :mod:`torch`, :mod:`torch.nn`,
 :mod:`torch.autograd`, and writing custom C extensions utilizing our C
 libraries.
 
+Extending :mod:`torch`
+----------------------
+
+The functions in the :mod:`torch` namespace can be overrided with custom
+user-specified implementations via the `__torch_function__` protocol introduced
+in PyTorch 1.4. Custom Python types with a `__torch_function__` method will
+bypass the standard PyTorch dispatch for :class:`Tensor` types. This works with
```

Comment:
I'm not sure "bypass the standard PyTorch dispatch for Tensor types" is going to mean much to the average reader. Maybe something like, "Tensor-like objects which define `__torch_function__` will have `__torch_function__` invoked when they are passed to functions in the torch namespace."